arlas-core/src/main/java/io/arlas/server/services/ExploreServices.java

```diff
@@ -147,6 +148,67 @@ public SearchResponse aggregate(MixedRequest request, CollectionReference collec
         return fluidSearch.exec();
     }
 
+    public ComputationResponse compute(MixedRequest request, CollectionReference collectionReference) throws ArlasException{
+        CheckParams.checkComputationRequest(request.basicRequest);
+        FluidSearch fluidSearch = new FluidSearch(client);
+        fluidSearch.setCollectionReference(collectionReference);
+        applyFilter(collectionReference.params.filter, fluidSearch);
+        applyFilter(request.basicRequest.filter, fluidSearch);
+        applyFilter(request.headerRequest.filter, fluidSearch);
+        String field = ((ComputationRequest)request.basicRequest).field;
+        ComputationEnum metric = ((ComputationRequest)request.basicRequest).metric;
+        fluidSearch = fluidSearch.compute(field, metric);
+        SearchResponse response;
+        try {
+            response = fluidSearch.exec();
+        } catch (SearchPhaseExecutionException e) {
+            throw new InvalidParameterException("The field's type must be numeric");
```

Comment:
It seems strange to catch a generic exception and translate it into a very specific exception. We might miss some cases and report to the user an entirely unrelated error. Shouldn't we provide the e.getMessage() ?

Reply:
Fixed